avro/src/main/java/com/fasterxml/jackson/dataformat/avro/deser/AvroReaderFactory.java

```diff
@@ -36,22 +38,23 @@
 
     /**
      * Method for creating a reader instance for specified type.
+     * @param decoder 
      */
-    public AvroStructureReader createReader(Schema schema)
+    public AvroStructureReader createReader(Schema schema, ResolvingDecoder decoder)
     {
         AvroStructureReader reader = _knownReaders.get(_typeName(schema));
         if (reader != null) {
             return reader;
         }
         switch (schema.getType()) {
         case ARRAY:
-            return createArrayReader(schema);
```

Comment:
Ok, so potential problem here is that so far readers have been stateless and reusable (prototype version has, anyway). I think this becomes actual problem too (unless something prevents it as side-effect... avro has various check), since although only `BinaryDecoder` has reference to physical input, `ResolvingDecoder` then delegates to a binary decoder.